src/Network/Network/VirtualNetwork/Subnet/ResetAzureVirtualNetworkSubnetPolicyCommand.cs

```diff
@@ -0,0 +1,109 @@
+    public class ResetAzureVirtualNetworkSubnetPolicyCommand : VirtualNetworkBaseCmdlet
+    {
+        private const string DefaultParameterSet = "SetByResourceId";
+        private const string VirtualNetworkParameterSet = "VirtualNetworkParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = VirtualNetworkParameterSet,
+            HelpMessage = "The name of the subnet")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "The name of the subnet")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = VirtualNetworkParameterSet,
+            HelpMessage = "Name of the service deletion. Use Get-AzureRmAvailableServiceDelegation command to list of ServiceName")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "Name of the service deletion. Use Get-AzureRmAvailableServiceDelegation command to list of ServiceName")]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Alias("InputObject")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = VirtualNetworkParameterSet,
+            ValueFromPipeline = true,
+            HelpMessage = "The virtualNetwork object")]
+        public PSVirtualNetwork VirtualNetwork { get; set; }
+
+        [Parameter(
+            ParameterSetName = DefaultParameterSet,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The resource Id of the virtual network resource")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        public override void Execute()
+        {
+            base.Execute();
+
+            if (this.IsParameterBound(c => c.ResourceId) || !string.IsNullOrWhiteSpace(this.ResourceId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId.Trim('"'));
+                var resourceGroupName = resourceIdentifier.ResourceGroupName;
+                var virtualNetworkName = resourceIdentifier.ResourceName;
+                this.VirtualNetwork = GetVirtualNetwork(resourceGroupName, virtualNetworkName);
+            }
+
+            // Verify if the subnet exists in the VirtualNetwork
+            var subnet = this.VirtualNetwork.Subnets.SingleOrDefault(resource => string.Equals(resource.Name, this.Name, StringComparison.CurrentCultureIgnoreCase));
+            if (subnet == null)
+            {
+                throw new ArgumentException(Properties.Resources.SubnetWithTheSpecifiedNameDoesNotExist);
+            }
+
+            if(!this.ServiceName.Contains("/"))
+            {
+                throw new ArgumentException(Properties.Resources.ServiceNameFormatGuidance);
+            }
```

Comment:
Why do we need this validation? Is there similar validation for when subnet gets delegated in command: 
Add-AzureRmDelegation? We should let server have that validation.